xml/schema/Member/MembershipType.xml

```diff
@@ -115,7 +115,8 @@
   <field>
     <name>minimum_fee</name>
     <title>membership Type Minimum Fee</title>
-    <type>decimal</type>
+    <type>varchar</type>
+    <length>512</length>
```

Comment:
Posted this to dev list:
There are also currencies with more than two decimal places (https://en.wikipedia.org/wiki/ISO_4217#Active_codes) and bitcoin supports even more.

Now, the problem is that most amounts are stored in the DB as decimal(10,2) which means any precision greater than 2 decimal places is lost.  For this particular issue (CRM-20772) I am only proposing changing a single field in MembershipType.minimum_fee as that is the only one that impacts on this issue (civicrm_price_field_value.amount is where most monetary values end up and is stored as varchar(512) (though probably shouldn't be)).

What I would like to identify, is what is a sensible datatype that I should change minimum_fee to, and that we should standardise on in future for monetary amounts?

1. Most fields holding monetary amounts in the db are decimal(10,2), some are decimal(20,2), tax_rate is decimal(10,8).
1. We should probably change civicrm_price_field_value.amount from varchar(512) to something more sensible - but that's probably too risky a change at this stage.
2. We should define a datatype that should be used to hold monetary amounts that supports more than the current two decimals.
3. The current datatype decimal(10,2) supports a maximum value of ten million: 10,000,000.00
4. Decimal(30,10) would support 10,000,000,000,000,000.0000000000
5. MySQL stores 9 decimal places in 4 bytes, decimals and fractions separately (https://dev.mysql.com/doc/refman/5.7/en/storage-requirements.html) meaning multiples of 9 are the most sensible values to use.
5. Currencies support up to 4 decimal places, bitcoin(BTC) apparently support 8 decimal places.  So following GAAP accounting principals 10 decimal places gives the maximum required precision.  So, 9 or 18 for decimal places?
6. For decimals, 9 gives 100,000,000 but we currently store 10, 18 gives 100,000,000,000,000,000 (one hundred thousand million million) which should be plenty?

So: decimal(18,9)?